setup_root.md

```diff
@@ -1,65 +1,211 @@
+* [API with authenthication](https://github.com/Zendro-dev/graphiql-auth): An implementation of the GraphQL IDE with Zendro login
+
+You can check docker containers by:
+```
+$ docker ps
 ```
-It is also possible to [run the zendro services locally](https://zendro-dev.github.io/zendro_cli.html#start-zendro-service), however you might run into unexpected incompatibilities depending on your local system.
+
+You can check docker logs by:
 ```
-zendro start
+$ docker logs -f <container name>
 ```
 
-### Step 5 (optional): Customize your project
+> ***Please wait until logs indicate the app is running on XXXX port to access Zendro services.***
+
+In default config, the running containers will be on ports:
+
+* Keycloak: http://10.5.0.11:8081
+   * The default keycloak username is *admin* and the password is *admin*.
+
+  ![Keycloak example](figures/kc1.png)
+  ![Keycloak example](figures/kc2.png)
+
+* SPA: http://localhost:8080
+    * The default zendro username is *zendro-admin* and the password is *admin*.
+
+  ![spa example](figures/login.png)
+  ![spa example](figures/spa.png)
+
+* GraphQL API: http://localhost:3000/graphql
+
+  ![api example](figures/graphql.png )
+
+* GraphQL API with authenthication: http://localhost:7000
+    * The default zendro username is *zendro-admin* and the password is *admin*.
+
+  ![api example](figures/login.png)
+  ![api example](figures/graphiql.png)
+
+
+If you wish to modify the default ports, adjust next files:
+> [Stop Zendro Instance](#stop), modify files and [start Zendro Instance](#start) again.
+
+* ./docker-compose-prod.yml
+* ./docker-compose-dev.yml
+* ./single-page-app/.env.production
+* ./single-page-app/.env.development
+* ./single-page-app/package.json
+* ./graphql-server/.env
+* ./graphiql-auth/.env.development
+* ./graphiql-auth/.env.production
+* ./graphiql-auth/package.json
+
+Also, if you wish to modify docker containers name or docker services names, adjust next files:
+> [Stop Zendro Instance](#stop), modify files and [start Zendro Instance](#start) again.
+
+* ./docker-compose-prod.yml
+* ./docker-compose-dev.yml
+
+Moreover, if you wish to modify keycloak IP adjust:
+* ipv4_address
+* subnet
+
+in *./docker-compose-prod.yml* and *./docker-compose-dev.yml*.
+
+
+#### Without docker
+If you prefer to use local setup with Keycloak, there are a few things to do after running Zendro:
+
+***MISSING***
+
+* install java
+* Install keycloak
+* Env vars and where to find them in keycloak, screenshots.
+* logs
+
+
+### Step 7: Start up Zendro with access control 
+Zendro can be used checking access rights for every single GraphQL query received by the currently logged in user identified by the Token. The user is decoded and his roles are loaded to check his access rights. This step is carried out by the [NPM acl package](https://www.npmjs.com/package/acl). Respective access rights can and must be declared in the file `./graphql-server/acl_rules.js`.
+
```

Comment:
write in a gender neutral way, `his` could be replaced by other words, e.g. corresponding roles.
mention that use can have a default acl_rules.js file.